src/core/pile.sol

```diff
@@ -130,14 +130,14 @@ contract Pile is DSNote, TitleOwned {
 
 
     // recovery used for defaulted loans_
-    function recovery(uint loan, uint wad) public auth {
-        doRepay(loan, wad);
+    function recovery(uint loan, address usr, uint wad) public auth {
```

Comment:
We need to pass the `usr` here because `msg.sender` would be the collector contract but we want to take the currency from the keeper